### PR TITLE
Adding event audit trail

### PR DESCRIPTION
**Pull request by buzztroll**
This PR adds functionality to track all blockade events that have been
performed on a blockade for its lifetime.  This allows users to reason
about the errors that blockade has teased out of their systems overtime.

### CODE REVIEW

File: blockade/cli.py [hunk 1]
```diff
@@ -472,6 +521,15 @@ def setup_parser():
             help="The maximum number of containers to effect at once.",
             type=int, default=1)
 
+    command_parsers["events"].add_argument(
+            "--json", action='store_true',
+            help="Show the data in JSON format.")
+    command_parsers["chaos"].add_argument(
```
Comment (labisso) [inline blockade/cli.py]: I think you meant to add this to the events command parser?

File: blockade/net.py [hunk 2]
```diff
@@ -13,13 +13,16 @@
...
+import traceback
 
 import docker
 
 import collections
 import itertools
 import re
 
+import sys
+
```
Comment (labisso) [inline blockade/net.py]: these new imports are unused

File: blockade/cli.py (hunk 1 above)
Comment (buzztroll) [inline blockade/cli.py]: I did

File: blockade/state.py [hunk 3]
```diff
@@ -186,3 +186,12 @@ def __write(self, containers, initialize=True):
             # clean up our created file
             self._state_delete()
             raise
+
+    def get_audit_file(self):
+        audit_dir = os.path.join(self._state_dir, "audit")
+        try:
+            os.makedirs(audit_dir, 0o755)
+        except OSError as os_e:
+            if os_e.errno != 17:
```
Comment (labisso) [inline blockade/state.py]: nitpick, but could you use `errno.EEXIST` here instead of the hardcoded 17

File: blockade/state.py [hunk 4]
```diff
@@ -186,3 +186,12 @@ def __write(self, containers, initialize=True):
...
+                raise
+        return os.path.join(audit_dir, "%s.json" % self._blockade_id)
```
Comment (labisso) [inline blockade/state.py]: what's the purpose of naming the file for the blockade? isn't there always going to be just one per directory?

File: blockade/cli.py [hunk 5]
```diff
@@ -359,6 +362,43 @@ def cmd_version(opts):
...
+    if opts.json:
+        outf = sys.stdout
+        if opts.output is not None:
+            outf = open(opts.output, "r")
+        try:
+            logs = b.get_audit().read_logs(as_json=False)
+            outf.write('{"events": [')
+            outf.write(os.linesep)
+            for l in logs:
+                outf.write(l)
```
Comment (labisso) [inline blockade/cli.py]: need a comma in between log lines

File: blockade/state.py (hunk 4 above)
Comment (buzztroll) [inline blockade/state.py]: It was not clear to me that was the case for daemon mode or for when --config was used.

**Pull request merged**